### ISSUE TITLE
Criteria for selecting the best model?

### ISSUE THREAD

**Issue by dariogonle** [ultralytics/yolov5#2857]
## ❔Question
Hi,

 I would like to known which is the criteria adopted to get the best.pt. I mean best.pt is the model where validation loss is minimized? Or it is necesarry to the validation loss increases it valua during a certain number of epochs?

Thank you.

**Comment by glenn-jocher** [comment]
@dariogonle best.pt is saved on every new best fitness epoch. fitness is defined a weighted combination of metrics in metrics.py which default to favoring mAP@0.5:0.95. You can modify the fitness function here to your own priorities if applicable.
https://github.com/ultralytics/yolov5/blob/238583b7d5c19029920d56c417c406c829569c75/utils/metrics.py#L12-L16

**Comment by glenn-jocher** [comment]
@dariogonle checkpointing logic in train.py is located here, and may also be adapted as required:
https://github.com/ultralytics/yolov5/blob/238583b7d5c19029920d56c417c406c829569c75/train.py#L390-L409

**Comment by dariogonle** [comment]
Thank you.

**Comment by Aoobi** [comment]
> @dariogonle best.pt is saved on every new best fitness epoch. fitness is defined a weighted combination of metrics in metrics.py which default to favoring mAP@0.5:0.95. You can modify the fitness function here to your own priorities if applicable.
> https://github.com/ultralytics/yolov5/blob/238583b7d5c19029920d56c417c406c829569c75/utils/metrics.py#L12-L16

Hi, sorry for digging up an old thread, but I am wondering what is the motivation behind setting mAP@0.5:0.95 as 90% of fitness? Sorry if this is dumb question but I am only just starting with deep learning. I see that Recall has a weight of 0. Isn't recall important as well? 
So if I set the recall weight to 0.9 and mAP@0.5:0.95 to 0.1, would that make model detect more, but I would also get more fail positives (it would detect something it shouldn't)? Do I understand that correctly? 

Thanks :)

**Comment by dariogonle** [comment]
AP or mAP is the area under the precision-recall curve, so if AP is high, recall will also be high.

**Comment by glenn-jocher** [comment]
@Aoobi Recall is a relative metric as it is a function of confidence. mAP is an absolute metrics as it is evaluated over all possible confidences.
